zql/Makefile

```diff
@@ -1,21 +1,53 @@
 deps = $(CURDIR)/deps
 
+PEGJS = $(deps)/bin/pegjs
+PIGEON = $(deps)/bin/pigeon
+GOIMPORTS = $(deps)/bin/goimports
+
+PEGJS_VERSION = $(shell test -x $(PEGJS) && $(PEGJS) --version)
+PEGJS_EXPECTED = 0.10.0
+PIGEON_VERSION = $(shell test -d $(deps) && cd $(deps); go list -m github.com/mna/pigeon || echo "")
+PIGEON_EXPECTED = v1.0.1-20190520151103-9fec3898cef8
+GOIMPORTS_VERSION = $(shell test -d $(deps) && cd $(deps); go list -m golang.org/x/tools || echo "")
+GOIMPORTS_EXPECTED = v0.0.0-20200204192400-7124308813f3
+
 all: zql.go zql.js
 
 run: all
 	go run ./main
 
-.PHONY: zql.go
-zql.go:
+$(PEGJS):
+ifeq (,$(findstring $(PEGJS_EXPECTED),$(PEGJS_VERSION)))
+	mkdir -p $(deps)
+	npm install --global --prefix $(deps) pegjs@0.10.0
+endif
+
+GODEPS :=
+ifeq (,$(findstring $(PIGEON_EXPECTED),$(PIGEON_VERSION)))
+  GODEPS += github.com/mna/pigeon@v1.0.1-0.20190520151103-9fec3898cef8
```

Comment:
Oh, maybe this could be the reason. It's maddening that the substring after `@` doesn't match `PIGEON_EXPECTED`.